Ok, I might not be interpreting nixpkgs-review output correctly, but anyway here is what I got about tensorflow:
```cpp
ERROR: /build/source/tensorflow/python/eager/BUILD:18:11: Compiling tensorflow/python/eager/pywrap_tfe_src.cc failed: (Exit 1): gcc failed: error executing command /nix/store/bxic6j2whyg3z4h2x>
In file included from tensorflow/python/eager/pywrap_tfe_src.cc:55:
./tensorflow/python/util/stack_trace.h: In static member function 'static tensorflow::StackTrace tensorflow::StackTrace::Capture(int)':
./tensorflow/python/util/stack_trace.h:64:55: error: 'PyThreadState' {aka 'struct _ts'} has no member named 'frame'; did you mean 'cframe'?
   64 |     const PyFrameObject* frame = PyThreadState_GET()->frame;
      |                                                       ^~~~~
      |                                                       cframe
./tensorflow/python/util/stack_trace.h:66:56: error: invalid use of incomplete type 'const PyFrameObject' {aka 'const struct _frame'}
   66 |     for (; i < limit && frame != nullptr; frame = frame->f_back, ++i) {
      |                                                        ^~
In file included from bazel-out/k8-opt/bin/external/local_config_python/python_include/Python.h:42,
                 from ./tensorflow/python/lib/core/numpy.h:35,
                 from ./tensorflow/python/eager/pywrap_tensor.h:21,
                 from tensorflow/python/eager/pywrap_tfe_src.cc:51:
bazel-out/k8-opt/bin/external/local_config_python/python_include/pytypedefs.h:22:16: note: forward declaration of 'PyFrameObject' {aka 'struct _frame'}
   22 | typedef struct _frame PyFrameObject;
      |                ^~~~~~
./tensorflow/python/util/stack_trace.h:67:37: error: invalid use of incomplete type 'const PyFrameObject' {aka 'const struct _frame'}
   67 |       PyCodeObject* code_obj = frame->f_code;
      |                                     ^~
bazel-out/k8-opt/bin/external/local_config_python/python_include/pytypedefs.h:22:16: note: forward declaration of 'PyFrameObject' {aka 'struct _frame'}
   22 | typedef struct _frame PyFrameObject;
      |                ^~~~~~
At global scope:
cc1plus: note: unrecognized command-line option '-Wno-unknown-warning' may have been intended to silence earlier diagnostics

--- a/pkgs/development/python-modules/jupyter-collaboration/default.nix
+++ b/pkgs/development/python-modules/jupyter-collaboration/default.nix
@@ -1,29 +1,33 @@
 { lib
 , buildPythonPackage
 , pythonOlder
-, fetchFromGitHub
+, fetchPypi
+, hatch-jupyter-builder
+, hatch-nodejs-version
 , hatchling
 , pythonRelaxDepsHook
+, jupyter-events
+, jupyter-server
 , jupyter-server-fileid
 , jupyter-ydoc
 , ypy-websocket
+, pytest-asyncio
 , pytest-jupyter
 , pytestCheckHook
 }:
 
 buildPythonPackage rec {
-  pname = "jupyter-server-ydoc";
-  version = "0.8.0";
+  pname = "jupyter-collaboration";
+  version = "1.0.1";
 
-  disabled = pythonOlder "3.7";
+  disabled = pythonOlder "3.8";
 
   format = "pyproject";
 
-  src = fetchFromGitHub {
-    owner = "jupyterlab";
-    repo = "jupyter_collaboration";
-    rev = "refs/tags/v${version}";
-    hash = "sha256-KLb7kU5jsj6ihGO6HU3Y7uF+0PcwKoQlqQAhtO0oaJw=";
+  src = fetchPypi {
+    pname = "jupyter_collaboration";
+    inherit version;
+    hash = "sha256-cf7BpF6WSoHQJQW0IXdpCAGTdkX9RNWZ4JovTHvcPho=";
   };
 
   postPatch = ''
@@ -31,6 +35,8 @@ buildPythonPackage rec {
   '';
 
   nativeBuildInputs = [
+    hatch-jupyter-builder
+    hatch-nodejs-version
     hatchling
     pythonRelaxDepsHook
   ];
@@ -40,14 +46,17 @@ buildPythonPackage rec {
   ];
 
   propagatedBuildInputs = [
+    jupyter-events
+    jupyter-server
     jupyter-server-fileid
     jupyter-ydoc
     ypy-websocket
   ];
 
-  pythonImportsCheck = [ "jupyter_server_ydoc" ];
+  pythonImportsCheck = [ "jupyter_collaboration" ];
 
   nativeCheckInputs = [
+    pytest-asyncio
     pytest-jupyter
     pytestCheckHook
   ];
@@ -57,10 +66,10 @@ buildPythonPackage rec {
   '';
 
   meta = {
-    changelog = "https://github.com/jupyterlab/jupyter_collaboration/blob/${src.rev}/CHANGELOG.md";
-    description = "A Jupyter Server Extension Providing Y Documents";
+    changelog = "https://github.com/jupyterlab/jupyter_collaboration/blob/v${version}/CHANGELOG.md";
+    description = "JupyterLab Extension enabling Real-Time Collaboration";
     homepage = "https://github.com/jupyterlab/jupyter_collaboration";
     license = lib.licenses.bsd3;
-    maintainers = with lib.maintainers; [ dotlambda ];
+    maintainers = lib.teams.jupyter.members;
   };
 }

--- a/pkgs/development/python-modules/jupyter-server/default.nix
+++ b/pkgs/development/python-modules/jupyter-server/default.nix
@@ -23,24 +23,27 @@
 , jupyter-server-terminals
 , nbformat
 , nbconvert
+, packaging
 , send2trash
 , terminado
 , prometheus-client
 , anyio
 , websocket-client
+, overrides
 , requests
+, flaky
 }:
 
 buildPythonPackage rec {
   pname = "jupyter-server";
-  version = "2.0.6";
+  version = "2.7.0";
   format = "pyproject";
-  disabled = pythonOlder "3.7";
+  disabled = pythonOlder "3.8";
 
   src = fetchPypi {
     pname = "jupyter_server";
     inherit version;
-    hash= "sha256-jddZkukLfKVWeUoe1cylEmPGl6vG0N9WGvV0qhwKAz8=";
+    hash = "sha256-NtoKJm0xpBrDNaNmyIkzwX36W7gXpI9cAsFtMDvJR38=";
   };
 
   nativeBuildInputs = [
@@ -60,22 +63,23 @@ buildPythonPackage rec {
     jupyter-server-terminals
     nbformat
     nbconvert
+    packaging
     send2trash
     terminado
     prometheus-client
     anyio
     websocket-client
+    overrides
   ];
 
   nativeCheckInputs = [
     ipykernel
-    pandoc
     pytestCheckHook
     pytest-console-scripts
     pytest-jupyter
     pytest-timeout
-    pytest-tornasync
     requests
+    flaky
   ];
 
   preCheck = ''
@@ -85,11 +89,16 @@ buildPythonPackage rec {
 
   disabledTests = [
     "test_cull_idle"
+    "test_server_extension_list"
   ] ++ lib.optionals stdenv.isDarwin [
     # attempts to use trashcan, build env doesn't allow this
     "test_delete"
     # test is presumable broken in sandbox
     "test_authorized_requests"
+    # Insufficient access privileges for operation
+    "test_regression_is_hidden"
+  ] ++ lib.optionals (stdenv.isLinux && stdenv.isAarch64) [
+    "test_copy_big_dir"
   ];
 
   disabledTestPaths = [
@@ -103,9 +112,10 @@ buildPythonPackage rec {
   __darwinAllowLocalNetworking = true;
 
   meta = with lib; {
+    changelog = "https://github.com/jupyter-server/jupyter_server/blob/v${version}/CHANGELOG.md";
     description = "The backend—i.e. core services, APIs, and REST endpoints—to Jupyter web applications";
     homepage = "https://github.com/jupyter-server/jupyter_server";
     license = licenses.bsdOriginal;
-    maintainers = [ maintainers.elohmeier ];
+    maintainers = lib.teams.jupyter.members;
   };
 }

--- a/pkgs/development/python-modules/jupyter-ydoc/default.nix
+++ b/pkgs/development/python-modules/jupyter-ydoc/default.nix
@@ -11,14 +11,14 @@
 
 buildPythonPackage rec {
   pname = "jupyter-ydoc";
-  version = "0.3.4";
+  version = "1.0.2";
 
   format = "pyproject";
 
   src = fetchPypi {
     pname = "jupyter_ydoc";
     inherit version;
-    hash = "sha256-WiJi5wvwBLgsxs5xZ16TMKoFj+MNsuh82BJa1N0a5OE=";
+    hash = "sha256-D5W+3j8eCB4H1cV8A8ZY46Ukfg7xiIkHT776IN0+ylM=";
   };
 
   nativeBuildInputs = [
@@ -46,6 +46,6 @@ buildPythonPackage rec {
     description = "Document structures for collaborative editing using Ypy";
     homepage = "https://github.com/jupyter-server/jupyter_ydoc";
     license = lib.licenses.bsd3;
-    maintainers = with lib.maintainers; [ dotlambda ];
+    maintainers = lib.teams.jupyter.members;
   };
 }

--- a/pkgs/top-level/python-aliases.nix
+++ b/pkgs/top-level/python-aliases.nix
@@ -182,6 +182,7 @@ mapAliases ({
   jupyter_client = jupyter-client; # added 2021-10-15
   jupyter_core = jupyter-core; # added 2023-01-05
   jupyter_server = jupyter-server; # added 2023-01-05
+  jupyter-server-ydoc = jupyter-collaboration; # added 2023-07-18
   Kajiki = kajiki; # added 2023-02-19
   Keras = keras; # added 2021-11-25
   larynx-train = piper-train; # added 2023-06-09

--- a/pkgs/top-level/python-packages.nix
+++ b/pkgs/top-level/python-packages.nix
@@ -5485,6 +5485,8 @@ self: super: with self; {
 
   jupyter-client = callPackage ../development/python-modules/jupyter-client { };
 
+  jupyter-collaboration = callPackage ../development/python-modules/jupyter-collaboration { };
+
   jupyter-contrib-core = callPackage ../development/python-modules/jupyter-contrib-core { };
 
   jupyter-contrib-nbextensions = callPackage ../development/python-modules/jupyter-contrib-nbextensions { };
@@ -5506,8 +5508,6 @@ self: super: with self; {
   jupyter-server-fileid = callPackage ../development/python-modules/jupyter-server-fileid { };
 
   jupyter-server-terminals = callPackage ../development/python-modules/jupyter-server-terminals { };
-
-  jupyter-server-ydoc = callPackage ../development/python-modules/jupyter-server-ydoc { };
 
   jupyter-ui-poll = callPackage ../development/python-modules/jupyter-ui-poll { };
 

